tests/integration/utils.py

```diff
@@ -463,3 +465,168 @@ def _validate_expected_dep_file_contents(dep_contents_file: Path, output_dir: Pa
         dep_file = output_dir / "deps" / path
         assert dep_file.exists()
         assert dep_file.read_text() == expected_content
+
+
+ALL = object()
+# Individual files could be added to the set as well.
+paths_to_code = set((Path("cachi2"), Path("tests/integration")))
+supported_pms = set(list(resolver._package_managers) + list(resolver._dev_package_managers))
+
+
+def retrieve_changed_files_from_git(ref: str = "HEAD") -> tuple[Path, ...]:
+    repo = Repo(".", search_parent_directories=True)
+    try:
+        commit = repo.commit(ref)
+    except ValueError:
+        # This is supposed to happen on gate only. There
+        # the following check will succeed. It will raise another
+        # ValueError locally
+        Path(".").absolute().relative_to("/__w")
+        repo.config_writer("global").set_value("safe", "directory", "*").release()
```

Comment:
This is an abomination, I am very inclined to do something about it. I am out of good ideas ATM though.

Reply:
This problem is tied to the CI environments and so the code to work around it needs to live in the GH action. Can you try any of the following to see if it helps (it should): https://github.com/actions/checkout/issues/766

Reply:
Thank you, GH actions were exactly the piece I was missing!